### DISCUSSION TITLE
Strange results of grain reconstruction

### DISCUSSION THREAD

**Discussion by NikitaSakharov** [mtex-toolbox/mtex#878] [Ask Anything] [answered]
Hi everyone!

I've got some strange result, I can't understand. So could you please help me?

First I'm doing some standard operation (import ebsd data, calculate grains, calculate twins, merge grains), and here is the code.
```matlab
%now reconstruct grains
[grains_initial,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[2*degree, 15*degree]);
%delete all of them less then 5 pixels
ebsd(grains_initial(grains_initial.grainSize<5)) = [];
%and now re-reconstruct grains using corrected ebsd data
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[2*degree, 15*degree]);
%now let's work with the boundaries
%extract grain boundaries from all grains and smooth them
gB = grains.boundary;
smooth_const = 5;
grains = smooth(grains, smooth_const);
%define twinning rotation
%define crystal symmetry first
CS = ebsd('Copper').CS;
%define rotation direction
m = Miller(1, 1, 1, CS);
%and define rotation axis
rot = rotation.byAxisAngle(m, 60 * degree);
twinning = rot;
isTwinning = angle(gB.misorientation, twinning) < 5*degree;
%extract twins now
twinBoundary = gB(isTwinning);
[mergedGrains,parentId] = merge(grains,twinBoundary,'calcMeanOrientation');
%вычисляем площадь child-зерен и записываем их
%в соотв. массив some_const(i) такого же размера, что и размер mergedGrains
twins_param = zeros(1, mergedGrains.length);
for i = 1:mergedGrains.length
twins_param(i) = mean(grains(parentId == mergedGrains(i).id).area)/mergedGrains(i).area;
end
%
twinId = unique(gB(isTwinning).grainId);
grains_tmp = grains;
grains_tmp(twinId) = [];
nonTwinId = grains_tmp.id;
```
And now I'm plotting the grains using the following code
```matlab
plot(grains, grains.meanOrientation)
hold on
plot(gB, 'linecolor','black','linewidth',1)
plot(twinBoundary, 'linecolor','red','linewidth',1)
plot(grains(nonTwinId).boundary, 'linecolor','black','linewidth',2)
hold off
```
And I get the following figure (let it be figure #1) and everything seems OK. There are some untwinned little grains.

![fig1](https://user-images.githubusercontent.com/62562246/111895186-06a61180-8a22-11eb-87e6-e1906e0cc24e.jpg)

But then I'm plotting grains using 
```matlab
plot(mergedGrains,mergedGrains.meanOrientation)
```
and get the following figure (figure #2)

![fig2](https://user-images.githubusercontent.com/62562246/111895290-a5327280-8a22-11eb-9660-5fc74677a648.jpg)

And if I compare fig#1 and fig#2 I see something is wrong. What I thought I'll get when I plot merged grains is all the grains except non-twinned ones. But I see all of them on fig#2 in spite I see some of them are non-twinned on fig #1.

So what is going wrong?

**Comment by ralfHielscher** [comment, marked as answer]
Merge simply merges grains according to some condition. It does not single
out grains.

Ralf.

********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
***@***.***
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Reply by NikitaSakharov** [reply]
So if the required condition is not satisfied, it leaves the grain 'not touched'. Right?

**Comment by ralfHielscher** [comment]
exactly :)

********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
***@***.***
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Reply by NikitaSakharov** [reply]
Thanks, Ralf!
